ush/jediinc2fv3.py

```diff
@@ -57,22 +60,136 @@ def jedi_inc_to_fv3(FV3JEDIinc, FV3inc):
+            prsi_ges = np.zeros((nlevs+1, nlats, nlons), float)
+            prsi_anl = np.zeros((nlevs+1, nlats, nlons), float)
+            k = 0
+            while k < nlevs+1:
+                prsi_ges[k, :, :] = ak[k]+bk[k]*ps_ges[:, :]
+                prsi_anl[k, :, :] = ak[k]+bk[k]*ps_anl[:, :]
+                k = k + 1
+
+            # Compute pressure increment (delp_inc)
+            # Compute guess and analysis layer pressures using Philips method
+            delp_inc = np.zeros((nlevs, nlats, nlons), float)
+            prsl_ges = np.zeros((nlevs, nlats, nlons), float)
+            prsl_anl = np.zeros((nlevs, nlats, nlons), float)
+            k = 0
+            while k < nlevs:
+                dbk = bk[k+1] - bk[k]
+                delp_inc[k, :, :] = ps_inc[:, :] * dbk
+                prsl_ges[k, :, :] = ((prsi_ges[k+1, :, :]**kap1 - prsi_ges[k, :, :]**kap1) / (kap1*(prsi_ges[k+1, :, :] - prsi_ges[k, :, :])))**kapr
+                prsl_anl[k, :, :] = ((prsi_anl[k+1, :, :]**kap1 - prsi_anl[k, :, :]**kap1) / (kap1*(prsi_anl[k+1, :, :] - prsi_anl[k, :, :])))**kapr
+                k = k + 1
+
+            # Write delp increment to output file
+            x = ncout.createVariable('delp_inc', 'f4', dimsout4)
+            ncout['delp_inc'][:] = delp_inc[:]
+
+            # Compute analysis temperature andl specific humidity
+            t_anl = np.zeros((nlevs, nlats, nlons), float)
+            q_anl = np.zeros((nlevs, nlats, nlons), float)
+            t_anl = np.add(t_ges, t_inc)
+            q_anl = q_ges + q_inc
+
+            # Compute guess and analysis virtual temperature
+            tv_ges = np.zeros((nlevs, nlats, nlons), float)
+            tv_anl = np.zeros((nlevs, nlats, nlons), float)
+            tv_ges = t_ges * (1.0 + fv*q_ges)
+            tv_anl = t_anl * (1.0 + fv*q_anl)
+
+            # Compute guess and analysis delz and delz increment
+            delz_ges = np.zeros((nlevs, nlats, nlons), float)
+            delz_anl = np.zeros((nlevs, nlats, nlons), float)
+            delz_inc = np.zeros((nlevs, nlats, nlons), float)
+            k = 0
+            while k < nlevs:
+                if k == 0:
+                    delz_ges[k, :, :] = rdog * tv_ges[:, k, :, :] * np.log(prsl_ges[k, :, :]/prsi_ges[k+1, :, :])
+                    delz_anl[k, :, :] = rdog * tv_anl[:, k, :, :] * np.log(prsl_anl[k, :, :]/prsi_anl[k+1, :, :])
+                else:
+                    delz_ges[k, :, :] = rdog * tv_ges[:, k, :, :] * np.log(prsi_ges[k, :, :]/prsi_ges[k+1, :, :])
+                    delz_anl[k, :, :] = rdog * tv_anl[:, k, :, :] * np.log(prsi_anl[k, :, :]/prsi_anl[k+1, :, :])
+
+                delz_inc[k, :, :] = delz_anl[k, :, :] - delz_ges[k, :, :]
+                k = k + 1
+
+            # Write delz increment to output file
+            x = ncout.createVariable('delz_inc', 'f4', dimsout4)
+            ncout['delz_inc'][:] = delz_inc[:]
+
     except FileNotFoundError as e:
```

Comment:
that's a lot of indented code after `try:`. Could your error trap be moved earlier in the code? Use `logging` instead of print?

Reply:
I replaced `print` with `logging`.   I encapsulated the generation of the fv3 increment file in a function.  There is still a fair amount of indentation in the script but at least the `try:except` section is greatly compacted.